cd /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/ && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/7_H9ES_P31_CEPT_d60_S61.trim_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/7_H9ES_P31_CEPT_d60_S61.trim_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/7_H9ES_P31_CEPT_d60_S61_hg38 && htseq-count -f bam -r pos -s no -t exon -m union -i gene_name /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/7_H9ES_P31_CEPT_d60_S61_hg38Aligned.sortedByCoord.out.bam /fdb/GENCODE/Gencode_human/release_27/gencode.v27.annotation.gtf > /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/htseq/7_H9ES_P31_CEPT_d60_S61_htseq_counts.txt
cd /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/ && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/8_H9ES_P31_CEPT_d60_S62.trim_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/8_H9ES_P31_CEPT_d60_S62.trim_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/8_H9ES_P31_CEPT_d60_S62_hg38 && htseq-count -f bam -r pos -s no -t exon -m union -i gene_name /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/8_H9ES_P31_CEPT_d60_S62_hg38Aligned.sortedByCoord.out.bam /fdb/GENCODE/Gencode_human/release_27/gencode.v27.annotation.gtf > /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/htseq/8_H9ES_P31_CEPT_d60_S62_htseq_counts.txt
cd /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/ && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/9_H9ES_P31_CEPT_d60_S63.trim_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Trimmed_FASTQs/9_H9ES_P31_CEPT_d60_S63.trim_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/9_H9ES_P31_CEPT_d60_S63_hg38 && htseq-count -f bam -r pos -s no -t exon -m union -i gene_name /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/Aligned_BAMs/9_H9ES_P31_CEPT_d60_S63_hg38Aligned.sortedByCoord.out.bam /fdb/GENCODE/Gencode_human/release_27/gencode.v27.annotation.gtf > /data/NCATS_ifx/data/mRNASeq/ISB013/sra_merge/htseq/9_H9ES_P31_CEPT_d60_S63_htseq_counts.txt
